containerfixture/containerfixture.go

```diff
@@ -0,0 +1,118 @@
+package containerfixture
+
+import (
+	_ "embed"
+	"fmt"
+	"log"
+	"math/rand"
+	"os/exec"
+	"strings"
+	"time"
+)
+
+//go:embed containerfixture.yaml
+var manifest string
+
+const Username = "admin"
+const Password = "Password"
+
+// Generates a random dynamic port number
+func GetRandomPort() string {
+	minPort := 49152
+	maxPort := 65535
+	r := rand.New(rand.NewSource(time.Now().UnixNano()))
+
+	randomPort := fmt.Sprintf("%d", r.Intn(maxPort-minPort)+minPort)
+	return randomPort
+}
+
+// Modify the manifest to allow running it in parallel as
+// many times as needed with dynamic values
+func modifyYaml(yaml string) map[string]string {
```

Comment:
GO has real structs, no reason to use maps in their place.

You could also just return multiple outputs.

Reply:
type DeploymentInfo struct {
	Yaml        string
	PodName     string
	WebPort     string
	ApiPort     string
}
```suggestion
func modifyYaml(yaml string) (*DeploymentInfo, error) {
```

Reply:
fixed